### PR TITLE
Aarnq/upgrade falco

### PR DESCRIPTION
**Pull request by aarnq**
**What this PR does / why we need it**:
Upgrades Falco and rewrites the rules to use upstream conventions and reuse the default rules.

**Which issue this PR fixes**:
Fixes: #1017

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:
I went through some of the rules to update them and reduce alerting for what is known, but I've not found a good way to target ingress-nginx since it doesn't report it's image repository.

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [x] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/falco-psp/values.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Default values for falco-psp.
```
Comment (Xartos) [inline helmfile/charts/falco-psp/values.yaml:1]: I guess this file isn't really correct, little too much unused variables

File: helmfile/50-applications.yaml [hunk 2]
```diff
@@ -203,6 +203,10 @@ releases:
   chart: ./upstream/falco
   version: 1.16.0
```
Comment (Xartos) [inline helmfile/50-applications.yaml:204]:
```suggestion
  version: 2.2.0
```

File: helmfile/values/falco.yaml.gotmpl [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
     enabled: true
 
-
 resources: {{- toYaml .Values.falco.resources | nindent 2  }}
 nodeSelector: {{- toYaml .Values.falco.nodeSelector | nindent 2  }}
 affinity: {{- toYaml .Values.falco.affinity | nindent 2  }}
 tolerations: {{- toYaml .Values.falco.tolerations | nindent 2  }}
 
-podSecurityPolicy:
-  create: true
```
Comment (Xartos) [inline helmfile/values/falco.yaml.gotmpl:22]: Why can't we use this one anymore?

File: helmfile/50-applications.yaml (hunk 2 above)
Comment (Xartos) [reply helmfile/50-applications.yaml:204]: I guess this should be updated on the exporter as well

**Review by raviranjanelastisys — APPROVED** [review]

File: helmfile/values/falco.yaml.gotmpl (hunk 3 above)
Comment (Xartos) [reply helmfile/values/falco.yaml.gotmpl:22]: Yea, I was looking at the wrong place apparently...

**Review by Xartos — APPROVED** [review]

**Pull request merged**